### PR TITLE
release: chore - add new release workflow using magefiles

### PR DESCRIPTION
**Pull request by iancardosozup**
This commit  updates the platform actual version in all dependent  files, add a new release process  with two new files and one refactor and the usage of magefile project https://magefile.org/.

In the magefile part added some privates commands in deployments/mage and imported some from horusec-devkit mageutils pkg

New files are release-beta.yml and release-rc.yml and refactor was made to new-release.yml now named release-new.yml to match the standards to be named only release.yml in a tested future.

Signed-off-by: Ian Cardoso <ian.cardoso@zup.com.br>

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by lucasbrunozup** [comment]
Didn't run the rest of the pipeline?

**Comment by iancardosozup** [comment]
> Didn't run the rest of the pipeline?

don't know why is not running, @matheusalcantarazup @wiliansilvazup @nathanmartinszup help?

File: deployments/mage/magefile.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+func UpdatePackageJson(version string) error {
+	if err := sh.RunV("npm", "install", "-g", "json"); err != nil {
+		return err
+	}
+	return sh.RunV("json", "-I", "-f", "./manager/package.json", "-e", `'this.version="`+version+`"'`)
+}
+
+func UpdateHorusecVersionInProject(actualVersion, releaseVersion string) error {
+	dir, _ := os.Getwd()
+	fmt.Println(dir)
```
Comment (matheusalcantarazup) [inline deployments/mage/magefile.go:40]: Debug println?

File: deployments/mage/magefile.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+//UpdatePackageJson updates packge.json lock horusec version
+func UpdatePackageJson(version string) error {
+	if err := sh.RunV("npm", "install", "-g", "json"); err != nil {
+		return err
+	}
+	return sh.RunV("json", "-I", "-f", "./manager/package.json", "-e", `'this.version="`+version+`"'`)
+}
+
+func UpdateHorusecVersionInProject(actualVersion, releaseVersion string) error {
+	dir, _ := os.Getwd()
```
Comment (matheusalcantarazup) [inline deployments/mage/magefile.go:39]: I think that would be good check this error

File: deployments/mage/magefile.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+	fmt.Println(dir)
+	return sh.RunV(
+		`find ./ -type f -not -path "./.git/*" -not -path "./Makefile" -not -path "./manager/cypress/*"  -not -name "*.sum" -not -name "*.mod"| xargs  sed -t -i "s/` + actualVersion + `/` + releaseVersion + `/g"`)
```
Comment (matheusalcantarazup) [inline deployments/mage/magefile.go:42]: Wouldn't be better use fmt.Sprintf here?

File: deployments/mage/magefile.go [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+	for k, v := range getConsingEnvs() {
+		if v == "" {
+			result = append(result, k)
+		}
+	}
+
+	if len(result) != 0 {
+		return fmt.Errorf("missing some env var: %v", result)
+	}
```
Comment (matheusalcantarazup) [inline deployments/mage/magefile.go:110]: Wouldn't be better return an error directly if v is empty? 
```go
for k, v := range getConsingEnvs() {
    if v == "" {
        return fmt.Errorf("missing env %s", k)
    }
}
```

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

**Comment by matheusalcantarazup** [comment]
> > Didn't run the rest of the pipeline?
> 
> don't know why is not running, @matheusalcantarazup @wiliansilvazup @nathanmartinszup help?

Try to close and reopen the PR. Something weird is happening in some PRs on github.

File: deployments/mage/magefile.go (hunk 1 above)
Comment (iancardosozup) [reply deployments/mage/magefile.go:40]: forgot it here, going to erase

File: deployments/mage/magefile.go (hunk 2 above)
Comment (iancardosozup) [reply deployments/mage/magefile.go:39]: this was a debug test gointo to erase

File: deployments/mage/magefile.go (hunk 4 above)
Comment (iancardosozup) [reply deployments/mage/magefile.go:110]: if we return on loop it won't print all missing envs

**Comment by lucasbrunozup** [comment]
I believe our workflows don't run for FORK.

**Review by matheusalcantarazup — APPROVED** [review]

**Comment by iancardosozup** [comment]
> I believe our workflows don't run for FORK.

you're right check [api workflow](https://github.com/ZupIT/horusec-platform/blob/b16908977e5eef27a1296baa71d9b593a90a701f/.github/workflows/api-pipeline.yml#L16) only runs on pushs, i will open another PR then rebase this

**Review by nathanmartinszup — APPROVED** [review]

**Comment by nathanmartinszup** [comment]
I really think that we should delete the old and new release files, this way is kinda messy.

**Pull request merged**